src/target/drivers/serial/usb_cdc/usb_uart.c

```diff
@@ -413,3 +413,39 @@ void usb_vcp_init(void) {
 
     nvic_enable_irq(NVIC_OTG_FS_IRQ);
 }
+
+// This is a crude spin-loop to wait until user has connected USB UART
+// To ensure that all messages are displayed
+static inline void wait_for_uart()
+{
+    Initialize_ButtonMatrix();
+    uint32_t lastms = CLOCK_getms();
+    while (1) {
+        u32 buttons = ScanButtons();
+        if (CHAN_ButtonIsPressed(buttons, BUT_EXIT))
+            break;
+        if (CLOCK_getms() - lastms < 250) {
+            continue;
+        }
+        lastms = CLOCK_getms();
+        printf("hello\n");
+        gpio_toggle(GPIOC, GPIO4);
+    }
+}
+
+void UART_Initialize()
```

Comment:
shall we consolidate usb initialize?

Reply:
probably.  I'm not sure how to best do that in this case, but if you have some idea of how it should look, let me know